test/github-url.ts

```diff
@@ -66,3 +66,16 @@ test('change filePath', t => {
 	t.is(url.href, 'https://github.com/microsoft/TypeScript/tree/master/package.json');
 	t.is(String(url), 'https://github.com/microsoft/TypeScript/tree/master/package.json');
 });
+
+test('get filePath from search', t => {
+	const url = new GitHubURL('https://github.com/yakov116/refined-github/commits/f23b687b3b89aa95a76193722cdfeff740646670?after=f23b687b3b89aa95a76193722cdfeff740646670+34&path%5B%5D=source&path%5B%5D=features&path%5B%5D=release-download-count.tsx');
+	t.is(url.user, 'yakov116');
+	t.is(url.repository, 'refined-github');
+	t.is(url.route, 'commits');
+	t.is(url.branch, 'f23b687b3b89aa95a76193722cdfeff740646670');
+	t.is(url.filePath, 'source/features/release-download-count.tsx');
+	t.is(url.pathname, '/yakov116/refined-github/commits/f23b687b3b89aa95a76193722cdfeff740646670/source/features/release-download-count.tsx');
+	t.is(url.href, 'https://github.com/yakov116/refined-github/commits/f23b687b3b89aa95a76193722cdfeff740646670/source/features/release-download-count.tsx');
+	t.is(url.search, '');
+	t.is(String(url), 'https://github.com/yakov116/refined-github/commits/f23b687b3b89aa95a76193722cdfeff740646670/source/features/release-download-count.tsx');
```

Comment:
> > > > > be more specific to `path[]`, probably.

You're changing the URL. They're not equivalent:

```js
// https://github.com/yakov116/refined-github/commits/f23b687b3b89aa95a76193722cdfeff740646670?after=f23b687b3b89aa95a76193722cdfeff740646670+34&path%5B%5D=source&path%5B%5D=features&path%5B%5D=release-download-count.tsx
// https://github.com/yakov116/refined-github/commits/f23b687b3b89aa95a76193722cdfeff740646670/source/features/release-download-count.tsx
```

Reply:
And I should be not? Or the test is wrong?

Reply:
I think this is the closest equivalent URL

```suggestion
	t.is(String(url), 'https://github.com/yakov116/refined-github/commits/f23b687b3b89aa95a76193722cdfeff740646670/source/features/release-download-count.tsx?after=f23b687b3b89aa95a76193722cdfeff740646670+34');
```

Reply:
https://github.com/yakov116/refined-github/blob/aefc57224678e5019380cb68d5742895dddfbea7/source/github-helpers/github-url.ts#L23-L25

```
			// Clear the search from the url, so it does not get passed to the new url.
			this.search = '';
			return {branch, filePath: filePathFromSearch};
```
So this is wrong correct?

Reply:
https://github.com/yakov116/refined-github/blob/aefc57224678e5019380cb68d5742895dddfbea7/source/github-helpers/github-url.ts#L23-L25

```
			// Clear the search from the url, so it does not get passed to the new url.
			this.search = '';
			return {branch, filePath: filePathFromSearch};
```
So this is wrong correct?

Reply:
The tests are correct now, so that is wrong since it drops the `after` parameter.